File: p4/src/java/Dosen.java
public class Dosen extends PegawaiTetap {
    private final double tunjanganFungsional;

    public Dosen(String nip, String nama, double gajiPokok, int masaKerja, double tunjanganFungsional) {
        super(nip, nama, gajiPokok, masaKerja);
        if (tunjanganFungsional < 0) {
            throw new IllegalArgumentException("Tunjangan fungsional tidak boleh negatif");
        }
        this.tunjanganFungsional = tunjanganFungsional;
    }
    @Override
    public double hitungGaji() {
        double gajiDasar = super.hitungGaji();
        return gajiDasar + tunjanganFungsional;
    }
    @Override
    public String jenis() {
        return "DOSEN";
    }
}

File: p4/src/php/Dosen.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/PegawaiTetap.php';

class Dosen extends PegawaiTetap
{
	public function __construct(
		string $nip, string $nama, float $gajiPokok,
		int $masaKerjaTahun,
		private readonly float $tunjanganFungsional,
	) {
		parent::__construct($nip, $nama, $gajiPokok, $masaKerjaTahun);
		if ($tunjanganFungsional < 0) {
			throw new InvalidArgumentException('Tunjangan fungsional tidak boleh negatif');
		}
	}

	public function hitungGaji(): float
	{
		return parent::hitungGaji() + $this->tunjanganFungsional;
	}

	public function jenis(): string { return 'DOSEN'; }
}
